### ISSUE TITLE
Analytics: Export data in ML-ready formats for predictive analytics

### ISSUE THREAD

**Issue by jpequegn** [jpequegn/trainrs#58]
## Overview
Export training data in machine learning-ready formats for external analysis and predictive modeling.

**Parent Issue:** #43 (Phase 4: Advanced FIT Features)
**Category:** Advanced Analytics Integration

## Scope
Prepare and export workout data optimized for machine learning workflows.

### Target ML Frameworks
- **Python/scikit-learn**: CSV with proper feature engineering
- **TensorFlow/PyTorch**: Time-series sequences in TFRecord/NPY
- **R/tidymodels**: Data frames with proper typing
- **Apache Spark**: Parquet format for big data

### Export Formats
- [ ] **CSV with engineered features**
  - Aggregated workout summaries
  - Rolling averages and trends
  - Normalized metrics
- [ ] **Time-series sequences**
  - NumPy arrays for deep learning
  - TFRecord format
  - HDF5 for large datasets
- [ ] **Parquet for big data**
  - Columnar storage
  - Efficient compression
  - Schema evolution support

### Feature Engineering
- [ ] Temporal features (day of week, season, etc.)
- [ ] Rolling statistics (7-day, 30-day averages)
- [ ] Training load metrics (CTL, ATL, TSB)
- [ ] Performance trends and derivatives
- [ ] Categorical encoding (sport, workout type)
- [ ] Normalization and scaling options

### Implementation Tasks
- [ ] Create ML export module
- [ ] Implement CSV export with features
- [ ] Support time-series export (NPY, TFRecord)
- [ ] Add Parquet export capability
- [ ] Feature engineering pipeline
- [ ] Data splitting utilities (train/val/test)
- [ ] Documentation for ML workflows

### Technical Details
```rust
pub struct MlExporter;

impl MlExporter {
    pub fn export_features_csv(
        workouts: &[Workout],
        path: &Path,
        config: &FeatureConfig
    ) -> Result<()>;
    
    pub fn export_timeseries_npy(
        workouts: &[Workout],
        path: &Path,
        sequence_length: usize
    ) -> Result<()>;
    
    pub fn export_parquet(
        workouts: &[Workout],
        path: &Path
    ) -> Result<()>;
}

pub struct FeatureConfig {
    pub include_temporal: bool,
    pub rolling_windows: Vec<u32>,
    pub normalize: bool,
    pub encode_categorical: bool,
}
```

### Use Cases
- Performance prediction
- Injury risk modeling
- Optimal training load forecasting
- Recovery time estimation
- Race performance prediction

### Acceptance Criteria
- [ ] Export to 3+ ML-ready formats
- [ ] Feature engineering pipeline
- [ ] Time-series sequence preparation
- [ ] Data splitting utilities
- [ ] Example Jupyter notebooks
- [ ] Documentation for ML workflows

**Priority:** Low
**Complexity:** Medium
**Estimated Effort:** 2-3 days
**Labels:** enhancement, analytics, machine-learning, export
